persist/sqlite/store.go

```diff
@@ -145,6 +158,120 @@ func (s *Store) clearLocks() error {
 	})
 }
 
+func sqlConn(ctx context.Context, db *sql.DB) (c *sqlite3.SQLiteConn, err error) {
+	if err := db.PingContext(ctx); err != nil {
+		return nil, fmt.Errorf("failed to ping database: %w", err)
+	}
+
+	raw, err := db.Conn(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create connection: %w", err)
+	}
+	err = raw.Raw(func(driverConn any) error {
+		var ok bool
+		c, ok = driverConn.(*sqlite3.SQLiteConn)
+		if !ok {
+			return errors.New("connection is not a SQLiteConn")
+		}
+		return nil
+	})
+	return
+}
+
+// backupDB is a helper function that creates a backup of the source database at
+// the specified path. The backup is created using the SQLite backup API, which
+// is safe to use with a live database.
+func backupDB(ctx context.Context, src *sql.DB, destPath string) (err error) {
+	// create the destination database
+	dest, err := sql.Open("sqlite3", sqliteFilepath(destPath))
+	if err != nil {
+		return fmt.Errorf("failed to open destination database: %w", err)
+	}
+	defer func() {
+		// errors are ignored
+		dest.Close()
+		if err != nil {
+			// remove the destination file if an error occurred during backup
+			os.Remove(destPath)
+		}
+	}()
+
+	// initialize the source conn
+	sc, err := sqlConn(ctx, src)
+	if err != nil {
+		return fmt.Errorf("failed to create source connection: %w", err)
+	}
+	defer sc.Close()
+
+	// initialize the destination conn
+	dc, err := sqlConn(ctx, dest)
+	if err != nil {
+		return fmt.Errorf("failed to create destination connection: %w", err)
+	}
+	defer dc.Close()
+
+	// start the backup
+	backup, err := dc.Backup("main", sc, "main")
```

Comment:
Considering that `dc` and `sc` are already connections, what are `main` and `main` used for?

Reply:
main is the default schema name in SQLite. So it’s backing up the main schema.